crates/core/tedge_agent/src/http_server/request_files.rs

```diff
@@ -88,5 +88,6 @@ fn local_path_for_file(
 }
 
 fn clean_utf8_path(path: &Utf8Path) -> Utf8PathBuf {
-    Utf8PathBuf::from(path_clean::clean(path.as_str()))
+    // unwrap is safe because clean returns an utf8 path when given an utf8 path
+    Utf8PathBuf::try_from(path_clean::clean(path.as_std_path())).unwrap()
```

Comment:
I'm assuming that we're using `path_clean::clean()` instead of `std::path::absolute()` because our MSRV is still 1.78 and the latter is only available from `1.79`.